tianshou/data/stats.py

```diff
@@ -0,0 +1,165 @@
+from collections.abc import MutableMapping
+from dataclasses import asdict, dataclass, field, InitVar
+
+import numpy as np
+
+
+# adapted from https://stackoverflow.com/questions/6027558/flatten-nested-dictionaries-compressing-keys
+def flatten_dict(dictionary: dict, parent_key='', separator='/',
+                 exclude_arrays: bool = True, exclude_none: bool = True):
+    items = []
+    for key, value in dictionary.items():
+        new_key = parent_key + separator + key if parent_key else key
+        if isinstance(value, MutableMapping):
+            items.extend(flatten_dict(value, new_key, separator=separator, exclude_arrays=exclude_arrays).items())
+        else:
+            if exclude_arrays and isinstance(value, np.ndarray | list):
+                continue
+            if exclude_none and value is None:
+                continue
+            items.append((new_key, value))
+    return dict(items)
+
+
+@dataclass(kw_only=True)
+class BaseStats(ABC):
+    """Abstract base class for statistics."""
+
+    def update(self, stats: dict):
+        for k, v in stats.items():
+            if np.isscalar(v):
+                setattr(self, k, v)
+            elif isinstance(v, np.ndarray | list):
+                setattr(self, k, ArrayStats(_array=v))
+            elif isinstance(v, BaseStats):
+                getattr(self, k).update(v.to_dict())
+
+    def to_dict(self, exclude: set[str] = None, exclude_arrays=True, exclude_none=True):
+        """Convert the dataclass to a dictionary.
+
+        :param exclude: set of field names to exclude from the dictionary.
+        :param exclude_arrays: if True, exclude arrays from the dictionary.
+        :param exclude_none: if True, exclude None from the dictionary.
+        :return: a dictionary of the dataclass instance
+        """
+        stat_dict = flatten_dict(asdict(self), exclude_arrays=exclude_arrays, exclude_none=exclude_none)
+        if exclude is not None:
+            stat_dict = {k: v for k, v in stat_dict.items() if k not in exclude}
+        return stat_dict
+
+
+@dataclass(kw_only=True)
+class ArrayStats(BaseStats):
+    """A data structure for storing the statistics of an array."""
+
+    _array: np.ndarray | list[float] = field(repr=False)
+    """The array to be analyzed."""
+    mean: float = field(init=False)
+    """The mean of the array."""
+    std: float = field(init=False)
+    """The standard deviation of the array."""
+    max: float = field(init=False)
+    """The maximum of the array."""
+    min: float = field(init=False)
+    """The minimum of the array."""
+
+    def __post_init__(self):
+        self.mean: float = np.mean(self._array)
+        """The mean of the array."""
+        self.std: float = np.std(self._array)
+        """The standard deviation of the array."""
+        self.max: float = np.max(self._array)
+        """The maximum of the array."""
+        self.min: float = np.min(self._array)
+        """The minimum of the array."""
```

Comment:
This is pretty nice! Small nit: remove the duplicated docstring since it has already represented above.